pandas/core/index.py

```diff
@@ -20,7 +20,30 @@
     ensure_index_from_sequences,
     get_objs_combined_axis,
 )
-from pandas.core.indexes.multi import _sparsify  # noqa:F401
+from pandas.core.indexes.multi import _sparsify
+
+__all__ = [
+    "CategoricalIndex",
+    "DatetimeIndex",
+    "Float64Index",
+    "Index",
+    "Int64Index",
+    "IntervalIndex",
+    "InvalidIndexError",
+    "MultiIndex",
+    "NaT",
+    "NumericIndex",
+    "PeriodIndex",
+    "RangeIndex",
+    "TimedeltaIndex",
+    "UInt64Index",
+    "_new_Index",
+    "ensure_index",
+    "ensure_index_from_sequences",
+    "get_objs_combined_axis",
+    "_sparsify",
```

Comment:
I don't think the private methods `_new_Index` and `_sparsify` should be in `__all__`. I guess they are being used by other modules, which shouldn't be the case.

I'm ok to merge it like this, if that's the case, but if you don't mind researching a bit, seeing where those are used, and adding at least a comment about this, that would be great.

I'd probably move these imports at the end, so they are separated from the rest.